### ISSUE TITLE
Add water quality sites in the map filter

### ISSUE THREAD

**Issue by Caesarh97** [aqualinkorg/aqualink-app#1026]
On the /map page where we have a drop-down list where we filter sites, I’d like to add all the sites with water quality data. Would it be easy to add these? In this filter, all sites that have either {sonde} or {hui} can be added. Both are displaying water quality data and would fit in this filter. This issue is of low priority, so if it’s fast and easy, we could add it now. Otherwise, it can wait. 

<img width="516" alt="Screenshot 2024-09-17 at 4 15 05 PM" src="https://github.com/user-attachments/assets/b158db68-9a99-4ae4-a3fe-366171af8bc6">

**Comment by yaodingyd** [comment]
Not sure if there's an easy way to do this. Right now we need to query `forcastData` and `latestData` two endpoints to find if there's any `hui` or `sonde` data, and these two endpoints require each specific site id, so we need to paginate through all sites and dispatch two endpoint for all of them to aggregate the collection of sites with water samplings. This seems rather efficient. 

I think to enable this feature, we would need to return
1)  water quality data to the top level `site` model, or
2) return ALL of latestData and forcastData
so we can process this in client side
not sure the implication/feasibility of doing this in api

**Comment by ericboucher** [comment]
Indeed, I think we probably need to update the API `/sites` endpoint to achieve this. We could maybe add a boolean flag to the site if it has received any water quality data in the past X weeks. Where `X weeks` is the same metric we currently used on the site page to determined if we will display Water Quality data or not. Is that something you feel comfortable looking into @yaodingyd ?

**Comment by yaodingyd** [comment]
sounds good, let me give it a go

**Comment by yaodingyd** [comment]
@ericboucher  is there a way to populate local db with some mock data? I'm doing development under `api`, applied the migrations but need some data (ideally medium scale) to run it end-to-end

**Comment by ericboucher** [comment]
@yaodingyd unfortunately we don't have a great way to do this at the moment, but I could create a small extract of our staging DB for you. Which tables would you need? You can email me: eric at ovio.org
